src/lib.rs

```diff
@@ -666,6 +676,51 @@ pub struct CameraAnimationOptions {
     animation_options: AnimationOptions,
 }
 
+#[derive(Debug)]
+struct SerdePathFindingError {
+    message: String,
+    path: Vec<String>,
+    raw_error: String,
+}
+
+impl fmt::Display for SerdePathFindingError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "Error in '{:?}': {}\n{}",
+            self.path, self.message, self.raw_error
+        )
+    }
+}
+
+impl From<serde_wasm_bindgen::Error> for SerdePathFindingError {
+    fn from(error: serde_wasm_bindgen::Error) -> Self {
+        let message = format!("{}", error);
+        let search_term = "::<impl serde::de::Deserialize for ";
+        let js_error: JsValue = error.into();
+        let mut path = vec![];
+        let mut raw_error = message.clone();
+
+        if let Some(e) = js_error.dyn_ref::<js_sys::Error>() {
+            raw_error = format!("{:?}", e);
+            for line in raw_error.lines() {
+                if let Some(start) = line.find(search_term) {
+                    let rest = &line[start + search_term.len()..];
+                    if let Some(end) = rest.find('>') {
+                        path.push(rest[..end].into())
+                    }
+                }
+            }
+        }
+        path = path.into_iter().rev().collect();
+        SerdePathFindingError {
+            message,
+            path,
+            raw_error,
+        }
+    }
+}
```

Comment:
The errors generated by wasm_bindgen where literally unintelligible. I made this super dum wrapper for debugging purposes and will delete it before we merge the PR :sweat_smile:.